### PR TITLE
can-873

### PR DESCRIPTION
**Pull request by array-junkie**
subscription details with breadcrumb api

### CODE REVIEW

File: app/Http/Controllers/CampController.php [hunk 1]
```diff
@@ -1027,11 +1027,20 @@ public function getCampBreadCrumb(Request $request, Validate $validate)
...
         try {
             $livecamp = Camp::getLiveCamp($filter);
-            $data = new stdClass();
             $data->bread_crumb = Camp::campNameWithAncestors($livecamp, $filter);
-            $indexs = ['bread_crumb'];
+            if ($request->user()) {
+                $campSubscriptionData = Camp::getCampSubscription($filter, $request->user()->id);
+                $data->flag = $campSubscriptionData['flag'];
+                $data->subscription_id = isset($campSubscriptionData['camp_subscription_data'][0]['subscription_id']) ? $campSubscriptionData['camp_subscription_data'][0]['subscription_id'] : null;
+                $data->subscribed_camp_name = isset($campSubscriptionData['camp_subscription_data'][0]['camp_name']) ? $campSubscriptionData['camp_subscription_data'][0]['camp_name'] : null;
```
Comment (akuks) [inline app/Http/Controllers/CampController.php:1041]: `$data->subscribed_camp_name = $campSubscriptionData['camp_subscription_data'][0]['camp_name'] ??  : null;`

How about this?

File: app/Http/Controllers/CampController.php [hunk 2]
```diff
@@ -1027,11 +1027,20 @@ public function getCampBreadCrumb(Request $request, Validate $validate)
...
+            }
+            $indexs = ['bread_crumb', 'flag', 'subscription_id', 'subscribed_camp_name'];
```
Comment (akuks) [inline app/Http/Controllers/CampController.php:1043]: Can you please move the indexes to the common file? I think we are already doing this

File: app/Models/Camp.php [hunk 3]
```diff
@@ -393,7 +393,7 @@ public static function getCampSubscription($filter, $userid = null): array
                 }
             }
             if (count($child_camps) > 0) {
-                $camp_subs_child = CampSubscription::where('user_id', '=', $userid)->whereIn('camp_num', $child_camps)->where('topic_num', '=', $filter['topicNum'])->where('subscription_start', '<=', strtotime(date('Y-m-d H:i:s')))->where('subscription_end', '=', null)->orWhere('subscription_end', '>=', strtotime(date('Y-m-d H:i:s')))->get();
+                $camp_subs_child = CampSubscription::where('user_id', '=', $userid)->whereIn('camp_num', $child_camps)->where('topic_num', '=', $filter['topicNum'])->where('subscription_start', '<=', strtotime(date('Y-m-d H:i:s')))->where('subscription_end', '=', null)->orWhere('subscription_end', '>', strtotime(date('Y-m-d H:i:s')))->get();
                 $flag = ($camp_subs_child && sizeof($camp_subs_child) > 0);
                 if ($flag) {
                     $flag = 2;
```
Comment (akuks) [inline app/Models/Camp.php:399]: What is `2` is here? Can you please update it in the comments

**Review by akuks — APPROVED** [review]

**Review by akuks — CHANGES_REQUESTED** [review]
Please see the comments. Code also refactored

File: app/Models/Camp.php (hunk 3 above)
Comment (M-Darab) [reply app/Models/Camp.php:399]: Flag 2 means that a child of a camp is subscribed. flag 1 means that camp is directly subscribed

File: app/Http/Controllers/CampController.php (hunk 2 above)
Comment (M-Darab) [reply app/Http/Controllers/CampController.php:1043]: All the indexes are different from each other. so that is why we have to define them separately.

**Review by akuks — APPROVED** [review]

**Pull request merged**